app/react/Uploads/actions/uploadsActions.js

```diff
@@ -93,42 +93,42 @@ export function upload(docId, file, endpoint = 'upload') {
   });
 }
 
-export function publicSubmit(data) {
-  return dispatch => new Promise((resolve, reject) => {
-    const request = superagent.post(`${APIURL}public`)
+export function publicSubmit(data, remote = false) {
+  return dispatch => new Promise((resolve) => {
+    const request = superagent.post(remote ? `${APIURL}remotepublic` : `${APIURL}public`)
     .set('Accept', 'application/json')
     .set('X-Requested-With', 'XMLHttpRequest')
     .field('captcha', data.captcha);
-    delete data.captcha;
 
     if (data.file) {
       request.attach('file', data.file);
-      delete data.file;
     }
 
     if (data.attachments) {
       data.attachments.forEach((attachment, index) => {
         request.attach(`attachments[${index}]`, attachment);
-        delete data.attachments;
       });
     }
-
-    request.field('entity', JSON.stringify(data));
-
+    request.field('entity', JSON.stringify(Object.assign({}, { title: data.title, template: data.template, metadata: data.metadata })));
+    let completionResolve;
+    let completionReject;
+    const uploadCompletePromise = new Promise((_resolve, _reject) => { completionResolve = _resolve; completionReject = _reject; });
     request
+    .on('progress', () => {
+      resolve({ promise: uploadCompletePromise });
+    })
     .on('response', (response) => {
       if (response.status === 200) {
         dispatch(notify('Success', 'success'));
-        resolve(response);
+        completionResolve(response);
         return;
       }
-
-      reject(response);
       if (response.status === 403) {
         dispatch(notify('Captcha error', 'danger'));
+        completionResolve(response);
```

Comment:
Hey @konzz what was the reason behind using `completionResolve(response)` instead of `completionReject(response)` when there's a 403 error?

I want to be able to detect failures in the `PublicForm` to prevent resetting the form when an error occurs, and I want to use `completionReject()` here so that I can bed caught be handled by the error callback of the promise. But I want to know whether there was an intentional reason here before I make the change.

Reply:
I remember doing it to detect when it was a captcha error and not reset the form, but if it's not working that way then I think you can change it.

Reply:
Thanks for the clarification. Then I will go ahead and change it.